In [1]:
from sklearn.preprocessing import StandardScaler
import numpy as np
import seaborn as sns
import matplotlib as plt
import pandas as pd
import matplotlib.pyplot as plt
plt.style.use('ggplot')
%matplotlib inline
import sklearn as sklearn
from scipy import stats
import statsmodels.api as sm
from sklearn.metrics import mean_absolute_error
from sklearn.linear_model import Ridge,Lasso,RidgeCV, LassoCV, ElasticNet, ElasticNetCV, LinearRegression
from sklearn.metrics import r2_score as r2
from sklearn.preprocessing import StandardScaler
from statsmodels.stats.outliers_influence import variance_inflation_factor 

import os
os.chdir(r'C:\Users\Shyam Adsul\Python codes SSPU\Data sets')


In [2]:
df=pd.read_csv('horse.csv')

In [3]:
df.head()

surgery    age  hospital_number  rectal_temp  pulse  respiratory_rate  \
0      no  adult           530101         38.5   66.0              28.0   
1     yes  adult           534817         39.2   88.0              20.0   
2      no  adult           530334         38.3   40.0              24.0   
3     yes  young          5290409         39.1  164.0              84.0   
4      no  adult           530255         37.3  104.0              35.0   

  temp_of_extremities peripheral_pulse mucous_membrane capillary_refill_time  \
0                cool          reduced             NaN            more_3_sec   
1                 NaN              NaN   pale_cyanotic            less_3_sec   
2              normal           normal       pale_pink            less_3_sec   
3                cold           normal   dark_cyanotic            more_3_sec   
4                 NaN              NaN   dark_cyanotic            more_3_sec   

   ... packed_cell_volume total_protein abdomo_appearance abdomo_protein  \
0  ...               45.0           8.4               NaN            NaN   
1  ...               50.0          85.0            cloudy            2.0   
2  ...               33.0           6.7               NaN            NaN   
3  ...               48.0           7.2     serosanguious            5.3   
4  ...               74.0           7.4               NaN            NaN   

      outcome  surgical_lesion lesion_1 lesion_2  lesion_3  cp_data  
0        died               no    11300        0         0       no  
1  euthanized               no     2208        0         0       no  
2       lived               no        0        0         0      yes  
3        died              yes     2208        0         0      yes  
4        died               no     4300        0         0       no  

[5 rows x 28 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 299 entries, 0 to 298
Data columns (total 28 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   surgery                299 non-null    object 
 1   age                    299 non-null    object 
 2   hospital_number        299 non-null    int64  
 3   rectal_temp            239 non-null    float64
 4   pulse                  275 non-null    float64
 5   respiratory_rate       241 non-null    float64
 6   temp_of_extremities    243 non-null    object 
 7   peripheral_pulse       230 non-null    object 
 8   mucous_membrane        252 non-null    object 
 9   capillary_refill_time  267 non-null    object 
 10  pain                   244 non-null    object 
 11  peristalsis            255 non-null    object 
 12  abdominal_distention   243 non-null    object 
 13  nasogastric_tube       195 non-null    object 
 14  nasogastric_reflux     193 non-null    object 
 15  nasoga

In [5]:
df.isnull().sum()

surgery                    0
age                        0
hospital_number            0
rectal_temp               60
pulse                     24
respiratory_rate          58
temp_of_extremities       56
peripheral_pulse          69
mucous_membrane           47
capillary_refill_time     32
pain                      55
peristalsis               44
abdominal_distention      56
nasogastric_tube         104
nasogastric_reflux       106
nasogastric_reflux_ph    246
rectal_exam_feces        102
abdomen                  118
packed_cell_volume        29
total_protein             33
abdomo_appearance        165
abdomo_protein           198
outcome                    0
surgical_lesion            0
lesion_1                   0
lesion_2                   0
lesion_3                   0
cp_data                    0
dtype: int64

In [6]:
df.shape

(299, 28)

In [7]:
df.dtypes.value_counts()

object     17
float64     7
int64       4
dtype: int64

In [8]:
df.describe()

hospital_number  rectal_temp       pulse  respiratory_rate  \
count     2.990000e+02   239.000000  275.000000        241.000000   
mean      1.087733e+06    38.168619   72.000000         30.460581   
std       1.532032e+06     0.733744   28.646219         17.666102   
min       5.184760e+05    35.400000   30.000000          8.000000   
25%       5.289040e+05    37.800000   48.000000         18.000000   
50%       5.303010e+05    38.200000   64.000000         25.000000   
75%       5.347360e+05    38.500000   88.000000         36.000000   
max       5.305629e+06    40.800000  184.000000         96.000000   

       nasogastric_reflux_ph  packed_cell_volume  total_protein  \
count              53.000000          270.000000     266.000000   
mean                4.707547           46.307407      24.274436   
std                 1.982311           10.436743      27.364194   
min                 1.000000           23.000000       3.300000   
25%                 3.000000           38.000000       6.500000   
50%                 5.000000           45.000000       7.500000   
75%                 6.500000           52.000000      56.750000   
max                 7.500000           75.000000      89.000000   

       abdomo_protein      lesion_1     lesion_2     lesion_3  
count      101.000000    299.000000   299.000000   299.000000  
mean         3.039604   3659.709030    90.528428     7.387960  
std          1.967947   5408.472421   650.637139   127.749768  
min          0.100000      0.000000     0.000000     0.000000  
25%          2.000000   2111.500000     0.000000     0.000000  
50%          2.300000   2322.000000     0.000000     0.000000  
75%          3.900000   3209.000000     0.000000     0.000000  
max         10.100000  41110.000000  7111.000000  2209.000000

In [9]:
df.corr()

hospital_number  rectal_temp     pulse  \
hospital_number               1.000000     0.144118  0.379607   
rectal_temp                   0.144118     1.000000  0.218868   
pulse                         0.379607     0.218868  1.000000   
respiratory_rate              0.273350     0.268846  0.469368   
nasogastric_reflux_ph        -0.005577     0.228306  0.005796   
packed_cell_volume           -0.093508     0.065101  0.405660   
total_protein                -0.228843    -0.060817 -0.086533   
abdomo_protein               -0.044184     0.009343  0.020937   
lesion_1                      0.145397     0.062639  0.079332   
lesion_2                     -0.050433    -0.062876 -0.001899   
lesion_3                     -0.020950     0.029336  0.101412   

                       respiratory_rate  nasogastric_reflux_ph  \
hospital_number                0.273350              -0.005577   
rectal_temp                    0.268846               0.228306   
pulse                          0.469368               0.005796   
respiratory_rate               1.000000               0.127164   
nasogastric_reflux_ph          0.127164               1.000000   
packed_cell_volume             0.086394              -0.131044   
total_protein                 -0.090743              -0.717363   
abdomo_protein                -0.074056               0.539505   
lesion_1                       0.180250               0.089207   
lesion_2                      -0.020601                    NaN   
lesion_3                       0.144773                    NaN   

                       packed_cell_volume  total_protein  abdomo_protein  \
hospital_number                 -0.093508      -0.228843       -0.044184   
rectal_temp                      0.065101      -0.060817        0.009343   
pulse                            0.405660      -0.086533        0.020937   
respiratory_rate                 0.086394      -0.090743       -0.074056   
nasogastric_reflux_ph           -0.131044      -0.717363        0.539505   
packed_cell_volume               1.000000      -0.053075        0.152736   
total_protein                   -0.053075       1.000000       -0.517338   
abdomo_protein                   0.152736      -0.517338        1.000000   
lesion_1                        -0.068743      -0.181468       -0.032938   
lesion_2                        -0.077727       0.223892       -0.153246   
lesion_3                        -0.066180       0.066856       -0.104158   

                       lesion_1  lesion_2  lesion_3  
hospital_number        0.145397 -0.050433 -0.020950  
rectal_temp            0.062639 -0.062876  0.029336  
pulse                  0.079332 -0.001899  0.101412  
respiratory_rate       0.180250 -0.020601  0.144773  
nasogastric_reflux_ph  0.089207       NaN       NaN  
packed_cell_volume    -0.068743 -0.077727 -0.066180  
total_protein         -0.181468  0.223892  0.066856  
abdomo_protein        -0.032938 -0.153246 -0.104158  
lesion_1               1.000000  0.005966  0.005850  
lesion_2               0.005966  1.000000  0.269374  
lesion_3               0.005850  0.269374  1.000000

**There are 17 categorical features present in the dataset**

      There are too many features hence lets divide the catogerical and numerical features so that workflow will be easy and sophisticated.

In [10]:
obj_columns=[]
nonobj_columns=[]
for col in df.columns.values:
    if df[col].dtype=='object':
        obj_columns.append(col)
    else:
        nonobj_columns.append(col)
print(len(obj_columns)," Object Columns are \n",obj_columns,'\n')
print(len(nonobj_columns),"Non-object columns are \n",nonobj_columns)

data_obj=df[obj_columns]
data_nonobj=df[nonobj_columns]

17  Object Columns are 
 ['surgery', 'age', 'temp_of_extremities', 'peripheral_pulse', 'mucous_membrane', 'capillary_refill_time', 'pain', 'peristalsis', 'abdominal_distention', 'nasogastric_tube', 'nasogastric_reflux', 'rectal_exam_feces', 'abdomen', 'abdomo_appearance', 'outcome', 'surgical_lesion', 'cp_data'] 

11 Non-object columns are 
 ['hospital_number', 'rectal_temp', 'pulse', 'respiratory_rate', 'nasogastric_reflux_ph', 'packed_cell_volume', 'total_protein', 'abdomo_protein', 'lesion_1', 'lesion_2', 'lesion_3']


In [11]:
percent_missing = df.isnull().sum() * 100 / len(df)

print(percent_missing)

surgery                   0.000000
age                       0.000000
hospital_number           0.000000
rectal_temp              20.066890
pulse                     8.026756
respiratory_rate         19.397993
temp_of_extremities      18.729097
peripheral_pulse         23.076923
mucous_membrane          15.719064
capillary_refill_time    10.702341
pain                     18.394649
peristalsis              14.715719
abdominal_distention     18.729097
nasogastric_tube         34.782609
nasogastric_reflux       35.451505
nasogastric_reflux_ph    82.274247
rectal_exam_feces        34.113712
abdomen                  39.464883
packed_cell_volume        9.698997
total_protein            11.036789
abdomo_appearance        55.183946
abdomo_protein           66.220736
outcome                   0.000000
surgical_lesion           0.000000
lesion_1                  0.000000
lesion_2                  0.000000
lesion_3                  0.000000
cp_data                   0.000000
dtype: float64


### Columns to drop:-
    Numerical Columns with higher missing count-
    1. nasogastric_reflux_ph    82.274247
    2. abdomo_protein           66.220736
    
    Categorical Columns with higher missing count-
    1. abdomo_appearance        55.183946

In [12]:
data_nonobj=data_nonobj.drop(['hospital_number'], axis = 1) 

In [13]:
data_nonobj.describe()

rectal_temp       pulse  respiratory_rate  nasogastric_reflux_ph  \
count   239.000000  275.000000        241.000000              53.000000   
mean     38.168619   72.000000         30.460581               4.707547   
std       0.733744   28.646219         17.666102               1.982311   
min      35.400000   30.000000          8.000000               1.000000   
25%      37.800000   48.000000         18.000000               3.000000   
50%      38.200000   64.000000         25.000000               5.000000   
75%      38.500000   88.000000         36.000000               6.500000   
max      40.800000  184.000000         96.000000               7.500000   

       packed_cell_volume  total_protein  abdomo_protein      lesion_1  \
count          270.000000     266.000000      101.000000    299.000000   
mean            46.307407      24.274436        3.039604   3659.709030   
std             10.436743      27.364194        1.967947   5408.472421   
min             23.000000       3.300000        0.100000      0.000000   
25%             38.000000       6.500000        2.000000   2111.500000   
50%             45.000000       7.500000        2.300000   2322.000000   
75%             52.000000      56.750000        3.900000   3209.000000   
max             75.000000      89.000000       10.100000  41110.000000   

          lesion_2     lesion_3  
count   299.000000   299.000000  
mean     90.528428     7.387960  
std     650.637139   127.749768  
min       0.000000     0.000000  
25%       0.000000     0.000000  
50%       0.000000     0.000000  
75%       0.000000     0.000000  
max    7111.000000  2209.000000

In [14]:
data_nonobj.isnull().sum()

rectal_temp               60
pulse                     24
respiratory_rate          58
nasogastric_reflux_ph    246
packed_cell_volume        29
total_protein             33
abdomo_protein           198
lesion_1                   0
lesion_2                   0
lesion_3                   0
dtype: int64

In [15]:
for col in data_nonobj.columns.values:
    if (pd.isna(data_nonobj[col]).sum())>0:
            data_nonobj[col]=data_nonobj[col].fillna(data_nonobj[col].mean())

In [16]:
data_nonobj.isnull().sum()

rectal_temp              0
pulse                    0
respiratory_rate         0
nasogastric_reflux_ph    0
packed_cell_volume       0
total_protein            0
abdomo_protein           0
lesion_1                 0
lesion_2                 0
lesion_3                 0
dtype: int64

In [17]:
data_obj.isnull().sum()

surgery                    0
age                        0
temp_of_extremities       56
peripheral_pulse          69
mucous_membrane           47
capillary_refill_time     32
pain                      55
peristalsis               44
abdominal_distention      56
nasogastric_tube         104
nasogastric_reflux       106
rectal_exam_feces        102
abdomen                  118
abdomo_appearance        165
outcome                    0
surgical_lesion            0
cp_data                    0
dtype: int64

In [18]:
df1=data_obj[['outcome']]

In [19]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 299 entries, 0 to 298
Data columns (total 1 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   outcome  299 non-null    object
dtypes: object(1)
memory usage: 2.5+ KB


In [20]:
data_obj=data_obj.drop(['outcome'],axis=1)

In [21]:
data_obj.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 299 entries, 0 to 298
Data columns (total 16 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   surgery                299 non-null    object
 1   age                    299 non-null    object
 2   temp_of_extremities    243 non-null    object
 3   peripheral_pulse       230 non-null    object
 4   mucous_membrane        252 non-null    object
 5   capillary_refill_time  267 non-null    object
 6   pain                   244 non-null    object
 7   peristalsis            255 non-null    object
 8   abdominal_distention   243 non-null    object
 9   nasogastric_tube       195 non-null    object
 10  nasogastric_reflux     193 non-null    object
 11  rectal_exam_feces      197 non-null    object
 12  abdomen                181 non-null    object
 13  abdomo_appearance      134 non-null    object
 14  surgical_lesion        299 non-null    object
 15  cp_data                

In [22]:
data_obj.mode()

surgery    age temp_of_extremities peripheral_pulse mucous_membrane  \
0     yes  adult                cool           normal     normal_pink   

  capillary_refill_time       pain peristalsis abdominal_distention  \
0            less_3_sec  mild_pain  hypomotile                 none   

  nasogastric_tube nasogastric_reflux rectal_exam_feces        abdomen  \
0           slight               none            absent  distend_large   

  abdomo_appearance surgical_lesion cp_data  
0            cloudy             yes      no

In [23]:
for col in data_obj.columns.values:
    if (pd.isna(data_obj[col]).sum())>0:
        data_obj[col]=data_obj[col].fillna(data_obj[col].mode()[0])

In [24]:
data_obj.isnull().sum()

surgery                  0
age                      0
temp_of_extremities      0
peripheral_pulse         0
mucous_membrane          0
capillary_refill_time    0
pain                     0
peristalsis              0
abdominal_distention     0
nasogastric_tube         0
nasogastric_reflux       0
rectal_exam_feces        0
abdomen                  0
abdomo_appearance        0
surgical_lesion          0
cp_data                  0
dtype: int64

In [25]:
data_obj.head()

surgery    age temp_of_extremities peripheral_pulse mucous_membrane  \
0      no  adult                cool          reduced     normal_pink   
1     yes  adult                cool           normal   pale_cyanotic   
2      no  adult              normal           normal       pale_pink   
3     yes  young                cold           normal   dark_cyanotic   
4      no  adult                cool           normal   dark_cyanotic   

  capillary_refill_time          pain peristalsis abdominal_distention  \
0            more_3_sec  extreme_pain      absent               severe   
1            less_3_sec     mild_pain      absent               slight   
2            less_3_sec     mild_pain  hypomotile                 none   
3            more_3_sec     depressed      absent               severe   
4            more_3_sec     mild_pain  hypomotile                 none   

  nasogastric_tube nasogastric_reflux rectal_exam_feces        abdomen  \
0           slight               none         decreased  distend_large   
1           slight               none            absent          other   
2           slight               none            normal         normal   
3             none       less_1_liter         decreased  distend_large   
4           slight               none            absent  distend_large   

  abdomo_appearance surgical_lesion cp_data  
0            cloudy              no      no  
1            cloudy              no      no  
2            cloudy              no     yes  
3     serosanguious             yes     yes  
4            cloudy              no      no

In [26]:
df1=df1.replace({'outcome':{'died':0,'euthanized':1,'lived':2}})

In [27]:
df_n = pd.get_dummies(data_obj,drop_first=True,dtype=np.int64)
data_obj = pd.concat([data_obj,df_n],axis=1)

In [28]:
data_obj = pd.concat([data_obj,df1],axis=1)

In [29]:
data_obj.head()

surgery    age temp_of_extremities peripheral_pulse mucous_membrane  \
0      no  adult                cool          reduced     normal_pink   
1     yes  adult                cool           normal   pale_cyanotic   
2      no  adult              normal           normal       pale_pink   
3     yes  young                cold           normal   dark_cyanotic   
4      no  adult                cool           normal   dark_cyanotic   

  capillary_refill_time          pain peristalsis abdominal_distention  \
0            more_3_sec  extreme_pain      absent               severe   
1            less_3_sec     mild_pain      absent               slight   
2            less_3_sec     mild_pain  hypomotile                 none   
3            more_3_sec     depressed      absent               severe   
4            more_3_sec     mild_pain  hypomotile                 none   

  nasogastric_tube  ... rectal_exam_feces_normal abdomen_distend_small  \
0           slight  ...                        0                     0   
1           slight  ...                        0                     0   
2           slight  ...                        1                     0   
3             none  ...                        0                     0   
4           slight  ...                        0                     0   

  abdomen_firm abdomen_normal abdomen_other abdomo_appearance_cloudy  \
0            0              0             0                        1   
1            0              0             1                        1   
2            0              1             0                        1   
3            0              0             0                        0   
4            0              0             0                        1   

   abdomo_appearance_serosanguious  surgical_lesion_yes  cp_data_yes  outcome  
0                                0                    0            0        0  
1                                0                    0            0        1  
2                                0                    0            1        2  
3                                1                    1            1        0  
4                                0                    0            0        0  

[5 rows x 57 columns]

In [30]:
data_obj.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 299 entries, 0 to 298
Data columns (total 57 columns):
 #   Column                            Non-Null Count  Dtype 
---  ------                            --------------  ----- 
 0   surgery                           299 non-null    object
 1   age                               299 non-null    object
 2   temp_of_extremities               299 non-null    object
 3   peripheral_pulse                  299 non-null    object
 4   mucous_membrane                   299 non-null    object
 5   capillary_refill_time             299 non-null    object
 6   pain                              299 non-null    object
 7   peristalsis                       299 non-null    object
 8   abdominal_distention              299 non-null    object
 9   nasogastric_tube                  299 non-null    object
 10  nasogastric_reflux                299 non-null    object
 11  rectal_exam_feces                 299 non-null    object
 12  abdomen               

In [31]:
data_nonobj.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 299 entries, 0 to 298
Data columns (total 10 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   rectal_temp            299 non-null    float64
 1   pulse                  299 non-null    float64
 2   respiratory_rate       299 non-null    float64
 3   nasogastric_reflux_ph  299 non-null    float64
 4   packed_cell_volume     299 non-null    float64
 5   total_protein          299 non-null    float64
 6   abdomo_protein         299 non-null    float64
 7   lesion_1               299 non-null    int64  
 8   lesion_2               299 non-null    int64  
 9   lesion_3               299 non-null    int64  
dtypes: float64(7), int64(3)
memory usage: 23.5 KB


In [32]:
data_merged = pd.concat([data_nonobj,data_obj],axis=1)

In [33]:
data_merged.head()

rectal_temp  pulse  respiratory_rate  nasogastric_reflux_ph  \
0         38.5   66.0              28.0               4.707547   
1         39.2   88.0              20.0               4.707547   
2         38.3   40.0              24.0               4.707547   
3         39.1  164.0              84.0               5.000000   
4         37.3  104.0              35.0               4.707547   

   packed_cell_volume  total_protein  abdomo_protein  lesion_1  lesion_2  \
0                45.0            8.4        3.039604     11300         0   
1                50.0           85.0        2.000000      2208         0   
2                33.0            6.7        3.039604         0         0   
3                48.0            7.2        5.300000      2208         0   
4                74.0            7.4        3.039604      4300         0   

   lesion_3  ... rectal_exam_feces_normal abdomen_distend_small abdomen_firm  \
0         0  ...                        0                     0            0   
1         0  ...                        0                     0            0   
2         0  ...                        1                     0            0   
3         0  ...                        0                     0            0   
4         0  ...                        0                     0            0   

  abdomen_normal abdomen_other abdomo_appearance_cloudy  \
0              0             0                        1   
1              0             1                        1   
2              1             0                        1   
3              0             0                        0   
4              0             0                        1   

  abdomo_appearance_serosanguious surgical_lesion_yes cp_data_yes outcome  
0                               0                   0           0       0  
1                               0                   0           0       1  
2                               0                   0           1       2  
3                               1                   1           1       0  
4                               0                   0           0       0  

[5 rows x 67 columns]

In [34]:
data_merged.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 299 entries, 0 to 298
Data columns (total 67 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   rectal_temp                       299 non-null    float64
 1   pulse                             299 non-null    float64
 2   respiratory_rate                  299 non-null    float64
 3   nasogastric_reflux_ph             299 non-null    float64
 4   packed_cell_volume                299 non-null    float64
 5   total_protein                     299 non-null    float64
 6   abdomo_protein                    299 non-null    float64
 7   lesion_1                          299 non-null    int64  
 8   lesion_2                          299 non-null    int64  
 9   lesion_3                          299 non-null    int64  
 10  surgery                           299 non-null    object 
 11  age                               299 non-null    object 
 12  temp_of_

In [35]:
X=data_merged.drop('outcome',axis=1)
y=data_merged[['outcome']]

In [36]:
X.head()

rectal_temp  pulse  respiratory_rate  nasogastric_reflux_ph  \
0         38.5   66.0              28.0               4.707547   
1         39.2   88.0              20.0               4.707547   
2         38.3   40.0              24.0               4.707547   
3         39.1  164.0              84.0               5.000000   
4         37.3  104.0              35.0               4.707547   

   packed_cell_volume  total_protein  abdomo_protein  lesion_1  lesion_2  \
0                45.0            8.4        3.039604     11300         0   
1                50.0           85.0        2.000000      2208         0   
2                33.0            6.7        3.039604         0         0   
3                48.0            7.2        5.300000      2208         0   
4                74.0            7.4        3.039604      4300         0   

   lesion_3  ... rectal_exam_feces_increased rectal_exam_feces_normal  \
0         0  ...                           0                        0   
1         0  ...                           0                        0   
2         0  ...                           0                        1   
3         0  ...                           0                        0   
4         0  ...                           0                        0   

  abdomen_distend_small abdomen_firm abdomen_normal abdomen_other  \
0                     0            0              0             0   
1                     0            0              0             1   
2                     0            0              1             0   
3                     0            0              0             0   
4                     0            0              0             0   

  abdomo_appearance_cloudy abdomo_appearance_serosanguious  \
0                        1                               0   
1                        1                               0   
2                        1                               0   
3                        0                               1   
4                        1                               0   

  surgical_lesion_yes cp_data_yes  
0                   0           0  
1                   0           0  
2                   0           1  
3                   1           1  
4                   0           0  

[5 rows x 66 columns]

In [37]:
X.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 299 entries, 0 to 298
Data columns (total 66 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   rectal_temp                       299 non-null    float64
 1   pulse                             299 non-null    float64
 2   respiratory_rate                  299 non-null    float64
 3   nasogastric_reflux_ph             299 non-null    float64
 4   packed_cell_volume                299 non-null    float64
 5   total_protein                     299 non-null    float64
 6   abdomo_protein                    299 non-null    float64
 7   lesion_1                          299 non-null    int64  
 8   lesion_2                          299 non-null    int64  
 9   lesion_3                          299 non-null    int64  
 10  surgery                           299 non-null    object 
 11  age                               299 non-null    object 
 12  temp_of_

In [38]:
X = X.select_dtypes(exclude=['object'])

In [39]:
X.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 299 entries, 0 to 298
Data columns (total 50 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   rectal_temp                       299 non-null    float64
 1   pulse                             299 non-null    float64
 2   respiratory_rate                  299 non-null    float64
 3   nasogastric_reflux_ph             299 non-null    float64
 4   packed_cell_volume                299 non-null    float64
 5   total_protein                     299 non-null    float64
 6   abdomo_protein                    299 non-null    float64
 7   lesion_1                          299 non-null    int64  
 8   lesion_2                          299 non-null    int64  
 9   lesion_3                          299 non-null    int64  
 10  surgery_yes                       299 non-null    int64  
 11  age_young                         299 non-null    int64  
 12  temp_of_

In [40]:
y.head()

outcome
0        0
1        1
2        2
3        0
4        0

In [41]:
from sklearn import preprocessing
le = preprocessing.LabelEncoder()

In [42]:
from sklearn.preprocessing import OrdinalEncoder
enc = OrdinalEncoder()

In [43]:
X = pd.DataFrame(enc.fit_transform(X),columns = X.columns)

In [44]:
y['outcome'].unique()

array([0, 1, 2], dtype=int64)

# Point to be noted-
#### After encoding
    died = 0
    euthanized = 1
    lived = 2

In [45]:
from sklearn.tree import DecisionTreeClassifier, export_graphviz
from sklearn import tree
from sklearn.model_selection import train_test_split, GridSearchCV
from sklearn.model_selection import cross_val_score
from sklearn.ensemble import RandomForestClassifier


In [46]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.30, random_state = 355)

In [49]:
#Random Forest Classifier¶
from sklearn.metrics import classification_report
rfc = RandomForestClassifier(n_estimators=200)
rfc.fit(X_train, y_train.values.ravel())
pred_rfc = rfc.predict(X_test)

In [50]:
print(classification_report(y_test, pred_rfc))


              precision    recall  f1-score   support

           0       0.68      0.68      0.68        19
           1       0.33      0.11      0.17         9
           2       0.82      0.90      0.86        62

    accuracy                           0.78        90
   macro avg       0.61      0.57      0.57        90
weighted avg       0.75      0.78      0.75        90



In [ ]:
#random forest got 0.75 % accuaracya

In [52]:
from sklearn.metrics import confusion_matrix
print(confusion_matrix(y_test, pred_rfc))


[[13  0  6]
 [ 2  1  6]
 [ 4  2 56]]


In [53]:
#SVM
from sklearn.svm import SVC


In [55]:
svc = SVC()
svc.fit(X_train, y_train.values.ravel())
pred_svc = svc.predict(X_test)

In [56]:
print(classification_report(y_test, pred_svc))


              precision    recall  f1-score   support

           0       0.69      0.47      0.56        19
           1       0.00      0.00      0.00         9
           2       0.77      0.95      0.85        62

    accuracy                           0.76        90
   macro avg       0.49      0.48      0.47        90
weighted avg       0.67      0.76      0.70        90



C:\Users\Shyam Adsul\anaconda3\lib\site-packages\sklearn\metrics\_classification.py:1221: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


In [57]:
#svm accuracy is 0.67